### ISSUE TITLE
Pre-built onborading request:  Valleysoft.DockerCredsProvider

### ISSUE THREAD

**Issue by vlada-shubina** [dotnet/source-build#3245]
### Describe the Problem

We are working on moving https://github.com/dotnet/sdk-container-builds to dotnet/sdk repo. 
Microsoft.NET.Build.Containers should become part of dotnet/sdk, and it used 3rd party package that we need to onboard to source-build: [Valleysoft.DockerCredsProvider](https://github.com/dotnet/sdk-container-builds/blob/54e9e6180f2360a9a10b38de3eb7cc3763e91152/Directory.Packages.props#L16)

### Describe the Solution

 [Valleysoft.DockerCredsProvider](https://github.com/dotnet/sdk-container-builds/blob/54e9e6180f2360a9a10b38de3eb7cc3763e91152/Directory.Packages.props#L16) is onboarded to source-build as pre-built.

### Additional Context

Please let me know if including it via source can make the process easier (as we are doing with wcwidth https://github.com/dotnet/source-build-reference-packages/tree/main/src/textOnlyPackages/src/wcwidth.sources/1.0.0)

**Comment by vlada-shubina** [comment]
@baronfel fyi

**Comment by omajid** [comment]
Thanks for filing this!

> [Valleysoft.DockerCredsProvider](https://github.com/dotnet/sdk-container-builds/blob/54e9e6180f2360a9a10b38de3eb7cc3763e91152/Directory.Packages.props#L16) is onboarded to source-build as pre-built.

Prebuilt are not allowed in source-build. source-build's entire reason for existence is to build the entire .NET product, end to end, from sources.

You can see more information about how we are working to eliminate prebuilts at https://github.com/dotnet/source-build/blob/main/Documentation/eliminating-pre-builts.md

It seems like https://www.nuget.org/packages/Valleysoft.DockerCredsProvider is from https://github.com/mthalman/docker-creds-provider? If so, I think we could add  as an additional repository that's built as part of the .NET build. https://github.com/dotnet/arcade/blob/main/Documentation/UnifiedBuild/VMR-Strategy-For-External-Source.md has some possible strategies that can be used in the source-build (VMR) context on how to carry these external sources.

cc @mthalman who maintains `Valleysoft.DockerCredsProvider` and is also a source-build dev.

**Comment by MichaelSimons** [comment]
Given https://github.com/mthalman/docker-creds-provider is not based on Arcade, I think this repo is a candidate to add to https://github.com/dotnet/source-build-externals.

**Comment by vlada-shubina** [comment]
> Given https://github.com/mthalman/docker-creds-provider is not based on Arcade, I think this repo is a candidate to add to https://github.com/dotnet/source-build-externals.

Would it be easier if we can do similar to `Wcwidth` scenario which is included via source?
I see it is included to source build via https://github.com/dotnet/source-build-reference-packages/tree/main/src/textOnlyPackages/src/wcwidth.sources/1.0.0

**Comment by vlada-shubina** [comment]
> You can see more information about how we are working to eliminate prebuilts at https://github.com/dotnet/source-build/blob/main/Documentation/eliminating-pre-builts.md
> 
> It seems like https://www.nuget.org/packages/Valleysoft.DockerCredsProvider is from https://github.com/mthalman/docker-creds-provider? If so, I think we could add as an additional repository that's built as part of the .NET build. https://github.com/dotnet/arcade/blob/main/Documentation/UnifiedBuild/VMR-Strategy-For-External-Source.md has some possible strategies that can be used in the source-build (VMR) context on how to carry these external sources.

I'm a bit confused here.
I followed https://github.com/dotnet/source-build/blob/main/Documentation/sourcebuild-in-repos/new-dependencies.md#adding-dependencies as a guideline , which has somewhat different proposals to https://github.com/dotnet/arcade/blob/main/Documentation/UnifiedBuild/VMR-Strategy-For-External-Source.md and are matching the ways that are in use at the moment.

According to check list this case goes either to 5 (if we can make a NuGet package with sources and compile them on the built-time like we do for wcwidth) or 7 (as @MichaelSimons suggested above).

**Comment by MichaelSimons** [comment]
>Would it be easier if we can do similar to Wcwidth scenario which is included via source?

No, wcwidth is allowed to be treated as a text-only package because the NuPkg only contains text.  This pattern is not acceptable for NuPkgs that contain binaries.

**Comment by omajid** [comment]
> According to check list this case goes either to 5 (if we can make a NuGet package with sources and compile them on the built-time like we do for wcwidth) or 7 (as @MichaelSimons suggested above).

Yeah, that's fair.

I would recommend 7 too. 5 has the potential for deferring what could be a build time error (and so easier to catch/fix) to a runtime error.

Is there any particular reason option 5 is more appealing (and was used for wcwidth)?

**Comment by vlada-shubina** [comment]
Forget to mention: we plan to move https://github.com/dotnet/sdk-container-builds to dotnet/sdk repo already for 7.0.300. 
I'm not sure if it matters for source-build though.

**Comment by MichaelSimons** [comment]
> Forget to mention: we plan to move https://github.com/dotnet/sdk-container-builds to dotnet/sdk repo already for 7.0.300. I'm not sure if it matters for source-build though.

Thanks for the heads up, yes it does matter as we are release source-build for each preview and therefore they must build without prebuilts.  This issue will need to be resolved to address the prebuilt.
